### ISSUE TITLE
Crashing upon leaveBreadcrumb completion block/crumb.metadata

### ISSUE THREAD

**Issue by jlafond-chwy** [bugsnag/bugsnag-cocoa#443]
### Description
I'm adding a breadcrumb from my Swift code and upon calling `Bugsnag.leaveBreadCrumb { crumb in ... }`, it crashes.

### Issue
Bugsnag's BugsnagBreadcrumb.m (line 89) in `- (NSDictionary *)objectValue` crashes due to an unrecognized selector for the object's copy functionality not being present.

### Environment
iPhone 11 Pro Max (13.3) simulator
iOS 13.3
Bugsnag 5.23.0

Library versions:

* Bugsnag version (from your Podfile, Podfile.lock, or elsewhere): 5.23.0
* CocoaPods version (if any) (`pod -v`): 1.7.3
* Carthage version (if any): N/A
* iOS/tvOS/macOS version(s) 13.3
* debug mode or production?: Debug
* How are you initializing Bugsnag? Can you share a snippet of your
  `AppDelegate` where Bugsnag is configured?

didFinishLaunchingWithOptions:  `Bugsnag.start(withApiKey: "e634d27b23743f20f55395024778520d")`

### Example code snippet

```swift
import Bugsnag

        var values = [Model.Value]()

        values.append(Model.Value("apple", value: 1, type: .primary))
        values.append(Model.Value("banana", value: 2, type: .secondary))

        let model = Model(name: "test", values: values, type: .head)

        guard let value = model.values.first else { return }

        Bugsnag.leaveBreadcrumb { crumb in

            crumb.name = "\(#file)::\(#function)"
            crumb.type = .log
            crumb.metadata = ["model": model,
                              "value": value,
                              "type": value.type]
        }

        Bugsnag.notifyError(NSError(domain: "Test", code: -1, userInfo: nil))
```

<details><summary>Error messages:</summary>

```
WARN : BSG_KSCrashSentry.c (103): BSG_KSCrashType bsg_kscrashsentry_installWithContext(BSG_KSCrash_SentryContext *, BSG_KSCrashType, void (*)(void *)): KSCrash: App is running in a debugger. Only user reported events will be handled.
2020-01-14 13:50:12.612260-0500 BugsnagCrash[36231:4800765] *** NSForwarding: warning: object 0x6000007a82d0 of class '_TtCV12BugsnagCrash5Model5Value' does not implement methodSignatureForSelector: -- trouble ahead
Unrecognized selector -[_TtCV12BugsnagCrash5Model5Value copy]
2020-01-14 13:50:12.612519-0500 BugsnagCrash[36231:4800765] Unrecognized selector -[_TtCV12BugsnagCrash5Model5Value copy]
(lldb) 
```
</details>

Attached project that will reproduce this:
[BugsnagCrash.zip](https://github.com/bugsnag/bugsnag-cocoa/files/4060883/BugsnagCrash.zip)

**Comment by mattdyoung** [comment]
Hi @Jp-Chewy 

Thanks for the report and the detailed reproduction case. Metadata added via `leaveBreadcrumb` needs to be serializable to JSON, but it shouldn't crash in the case that it's not.

As you hinted at in the issue above, the crash happens because the copy functionality isn't present in your `Value` class.

Inheriting from `NSObject`:
```
class Value : NSObject {
    ...
}
```
and implementing `NSCopying`:
```
extension Model.Value: NSCopying {
    enum ValueType {
        case primary, secondary
    }
    
    func copy(with zone: NSZone? = nil) -> Any {
        return Model.Value(key, value: value, type: type)
    }
}
```
prevents the crash and allows you to see the (expected) behaviour where you see the following output in the logs:
```
Unable to serialize breadcrumb: Not a valid JSON object
```

I'm raising this as a bug as we should protect against the crash.

Also, you should note that currently the only metadata which can be added to breadcrumbs is a flat list of key / value pairs (more complex structures are ignored). We are looking into improving this to allow the addition of more complex data types.

**Comment by Jp-Chewy** [comment]
Thank you, @mattdyoung. -Jp

**Comment by steve-nester-uk** [comment]
Fixed in [v6.1.0]( https://github.com/bugsnag/bugsnag-cocoa/releases/tag/) / PLAT-3862
